### ISSUE TITLE
YOLOv8n-obb algorithm on the DOTAv1 dataset shows an extremely low mAP for the "large vehicle" category.

### ISSUE THREAD

**Issue by QibangLi** [ultralytics/ultralytics#7238]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello! Thank you very much for your team's contributions in the field of oriented object detection. Currently, I also plan to further improve YOLOv8_obb based on your work. When I reproduced the YOLOv8n-obb algorithm on the DOTAv1 dataset, I found that the mAP for the "large vehicle" category is extremely low. Is this a normal phenomenon?
![image](https://github.com/ultralytics/ultralytics/assets/85171580/78d481b5-f0d6-4f96-bb4f-a8d712c0589e)
![image](https://github.com/ultralytics/ultralytics/assets/85171580/927a5d69-eef0-4b46-8e74-a3277c05d24a)

**Comment by QibangLi** [comment]
> 👋 Hello @QibangLi, thank you for your interest in Ultralytics YOLOv8 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
> 
> If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.
> 
> If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/).
> 
> Join the vibrant [Ultralytics Discord](https://ultralytics.com/discord) 🎧 community for real-time conversations and collaborations. This platform offers a perfect space to inquire, showcase your work, and connect with fellow Ultralytics users.
> 
> ## Install
> Pip install the `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).
> 
> ```shell
> pip install ultralytics
> ```
> 
> ## Environments
> YOLOv8 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
> 
> * **Notebooks** with free GPU: [![Run on Gradient](https://camo.githubusercontent.com/7c47520d799373030a5bc16f03365634982e82a5547167167cb1e00bb972d640/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://console.paperspace.com/github/ultralytics/ultralytics) [![Open In Colab](https://camo.githubusercontent.com/f5e0d0538a9c2972b5d413e0ace04cecd8efd828d133133933dfffec282a4e1b/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/1398db766d8ca60e6f296aac9ac429e344705c6c2c7e1ceb024230aac69fd6be/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/ultralytics/yolov8)
> * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/3aba317f5cc4948ed3007a4dcb5a4dfbdd499ca824993974b713eb1c0967104e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f756c7472616c79746963733f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/ultralytics)
> 
> ## Status
> [![Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule)
> 
> If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.



> 👋 Hello @QibangLi, thank you for your interest in Ultralytics YOLOv8 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
> 
> If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.
> 
> If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/).
> 
> Join the vibrant [Ultralytics Discord](https://ultralytics.com/discord) 🎧 community for real-time conversations and collaborations. This platform offers a perfect space to inquire, showcase your work, and connect with fellow Ultralytics users.
> 
> ## Install
> Pip install the `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).
> 
> ```shell
> pip install ultralytics
> ```
> 
> ## Environments
> YOLOv8 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
> 
> * **Notebooks** with free GPU: [![Run on Gradient](https://camo.githubusercontent.com/7c47520d799373030a5bc16f03365634982e82a5547167167cb1e00bb972d640/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://console.paperspace.com/github/ultralytics/ultralytics) [![Open In Colab](https://camo.githubusercontent.com/f5e0d0538a9c2972b5d413e0ace04cecd8efd828d133133933dfffec282a4e1b/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/1398db766d8ca60e6f296aac9ac429e344705c6c2c7e1ceb024230aac69fd6be/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/ultralytics/yolov8)
> * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/3aba317f5cc4948ed3007a4dcb5a4dfbdd499ca824993974b713eb1c0967104e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f756c7472616c79746963733f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/ultralytics)
> 
> ## Status
> [![Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule)
> 
> If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.


Thank you for your explanation. Currently, the most widely used dataset for target detection is the remote sensing dataset DOTA, which is divided into v1, v1.5, and v2 versions. I intend to reproduce YOLOv8-obb on the publicly available v1 version of the dataset. I downloaded dotav1.zip provided by the Ultralytics official website and used the data/split_dota.py script to split the dataset. I'm wondering if you could provide experimental results on the official DOTA dataset so that I can use them as a reference to improve my experiments. Additionally, the test portion of the DOTA dataset only provides images without publicly disclosed labels. To conduct tests, one needs to submit the labels in the specified format to the official website for testing. I'm curious if your team could provide a one-click script to facilitate test output on the DOTA dataset.
![image](https://github.com/ultralytics/ultralytics/assets/85171580/e24c7e88-3889-41c6-a234-0c522fb21fad)

**Comment by glenn-jocher** [comment]
👋 Hi @QibangLi, I appreciate your efforts in applying YOLOv8 to the DOTA dataset and your keen interest in improving oriented object detection.

For low mAP issues on certain categories like "large vehicle", it could be caused by several factors:
1. Class imbalance in the dataset.
2. Insufficient feature learning for the specific class.
3. Potential need for hyperparameter tuning.

To further assist you, please ensure that you're following our training best practices and making use of the full suite of YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) provided in the documentation.

As for providing official DOTA dataset results, we typically encourage members of the community to share and benchmark their own results as this fosters collaboration. For generating a submission file for the DOTA test set, while we do not have a one-click script available, the general process would involve using the Predict Mode to obtain results and then formatting them as required by the DOTA evaluation server.

If you have any more detailed observations from your training logs or specific queries about the steps you've taken, please share them to potentially pinpoint areas of improvement. Let's work together to make YOLOv8 even better! 🌟

Thanks for being a part of the YOLO community! 🤝

**Comment by Burhan-Q** [comment]
@QibangLi the OBB is still a work in progress. See ultralytics/ultralytics/pull/4499

> IMPORTANT: THIS PR IS STILL UNDER CONSTRUCTION AND OBB TRAINING DOES NOT WORK CORRECTLY YET.

**Comment by kaunghtetsan275** [comment]
Could you please provide the hyperparameters used for OBB training to reproduce the result?

**Comment by kaunghtetsan275** [comment]
Do you mean default.yaml? I can't seem to find hyp.scratch.yaml file.

**Comment by glenn-jocher** [comment]
@kaunghtetsan275 hi there! 😊 My apologies for the confusion. You're right, for YOLOv8, the hyperparameters are indeed specified in the `default.yaml` file, not `hyp.scratch.yaml`. You can find and adjust the hyperparameters directly in your `default.yaml` or specify them when running your training command. For example:

```bash
yolo train --data your_dataset.yaml --cfg yolov8n-obb.yaml --weights '' --hyp default.yaml
```

If you need to customize any hyperparameters, you can do so by editing the `default.yaml` file or passing them as arguments in the command line. Happy training! 🚀
